src/statsd_client.cpp

```diff
@@ -1,15 +1,42 @@
-
 #include <math.h>
-#include <netdb.h>
 #include <time.h>
-#include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
 #include <sys/types.h>
-#include <netinet/in.h>
 #include "statsd_client.h"
-#include <fcntl.h>
+
+/* platform-specific headers */
+#ifdef HAVE_WINSOCK2_H
+    #include <winsock2.h>
+#endif
+#ifdef HAVE_WS2TCPIP_H
+    #include <ws2tcpip.h>
+#endif
+
+#ifdef HAVE_SYS_SOCKET_H
+    #include <sys/socket.h>
+#endif
+
+#ifdef HAVE_APRA_INET_H
+  #include <arpa/inet.h>
+#endif
+
+#ifdef HAVE_NETDB_H
+  #include <netdb.h>  /* Needed for getaddrinfo() and freeaddrinfo() */
+#endif
+
+#ifdef HAVE_UNISTD_H
+    #include <unistd.h> /* Needed for close() */
+#endif
+
+#ifdef _WIN32
+    // Win32-specific close socket function
+    #define CLOSE_SOCKET(s) closesocket(s)
```

Comment:
I've dropped `check_function_exists` , because it wasn't able to find `closesocket` on windows. May be I need to add some additional headers etc., so I've moved it under `_WIN32`